bn-apps/ledger-sync/ledger-sync-service/src/main/kotlin/com/r3/businessnetworks/ledgersync/Helpers.kt

```diff
@@ -88,4 +77,22 @@ class ConfigurationService(appServiceHub : AppServiceHub): SingletonSerializeAsT
         }
     }
     open fun pageSize() = _config?.getInt("pageSize") ?: DEFAULT_PAGE_SIZE
+}
+
+fun ServiceHub.getRecycledTx(): List<SecureHash> {
+    if(!vrExist()) return listOf()
+    val list = mutableListOf<SecureHash>()
+    val hqlSelectQuery = "select tx_id from RECYCLABLE_TX as dbTable where dbTable.tx_id not in " +
+                            "(select nodeTXTable.TX_ID from NODE_TRANSACTIONS nodeTXTable) "
+    val result = jdbcSession().prepareStatement(hqlSelectQuery).executeQuery()
+    while(result.next()) list.add((SecureHash.parse(result.getString(1))))
+    return list
+}
+
+fun ServiceHub.vrExist(): Boolean {
+    val result = jdbcSession().prepareStatement("show tables").executeQuery()
+    while(result.next()) {
+        if (result.getString(1).equals("RECYCLABLE_TX")) return true
```

Comment:
Use `==` operator

Reply:
Minor points taken. HQL is exactly what I wanted to talked about. At the beginning I used HQL, but that seems to require some VR dependency. Then I tried this plain SQL, but new issue came up as it might not be suitable for certain backend DB.

Reply:
I do agree we should use HQL, but I don't think there is way to use it without VR dependency. Please enlighten me if there is such way.

Reply:
Let's move the HQL dicussion to the general conversation. It's not related to the `==` issue.